source/code/source/playfab/QoS/PlayFabQoSApi.cpp

```diff
@@ -287,7 +287,7 @@ namespace PlayFab
                 // Iterate over all the threads and servers that need to be pinged
                 for (size_t i = 0; i < numThreads && pingItr < numPings; ++i)
                 {
-                    if (asyncPingResults[i].valid()) // the very first ping result might be a fake future
+                    while (asyncPingResults[i].valid()) // the very first ping result might be a fake future
```

Comment:
I'm not super comfortable with an endless while.
I think it should be a for loop with a fixed-max number of loops so that we don't hang the thread (or process) in this loop.
(const value at the top of the file defining how max-loops used here)